src/algo/matrix_base/matrix_base.py

```diff
@@ -0,0 +1,33 @@
+from pathlib import Path
+
+from src.algo.algo_interface import CFPQAlgo
+
+from src.grammar.cnf_grammar import CnfGrammar
+from src.graph.label_graph import LabelGraph
+
+
+class MatrixBaseSolver(CFPQAlgo):
+    def __init__(self, path_to_graph: Path, path_to_grammar: Path):
+        super().__init__(path_to_graph, path_to_grammar)
+        self.graph = LabelGraph.from_txt(str(path_to_graph) + ".txt")
+        self.grammar = CnfGrammar.from_cnf(str(path_to_grammar) + ".cnf")
+
+    def solve(self):
```

Comment:
If you overwrite the ```solve``` method here, it is no longer abstract for class ```MatrixBaseSolver```  and for its descendants

Reply:
So, for example, now you can create an instance of ```MatrixBaseSolver```.